app/test/prepare_proposal_test.go

```diff
@@ -30,6 +33,74 @@ import (
 	blobtypes "github.com/celestiaorg/celestia-app/v4/x/blob/types"
 )
 
+func TestPrepareProposalValidConstruction(t *testing.T) {
+	// Reproduces https://github.com/celestiaorg/celestia-app/issues/4961
+	t.Run("prepare proposal creates a proposal that process proposal throws an account sequence mismatch", func(t *testing.T) {
+		encConf := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+		accounts := testfactory.GenerateAccounts(1)
+		testApp, kr := testutil.SetupTestAppWithGenesisValSetAndMaxSquareSize(app.DefaultConsensusParams(), 128, accounts...)
+		height := testApp.LastBlockHeight() + 1
+
+		txs := createBlobTxs(t, testApp, encConf, kr, accounts)
+		require.Equal(t, 9, len(txs))
+
+		prepareResponse, err := testApp.PrepareProposal(&abci.RequestPrepareProposal{
+			Txs:    txs,
+			Height: height,
+			Time:   time.Now(),
+		})
+		require.NoError(t, err)
+		require.Equal(t, 7, len(prepareResponse.Txs))
```

Comment:
[optional]

```suggestion
		require.NoError(t, err)
		// The filtered builder in prepare proposal should have dropped the last two txs.
		// The filtered builder should have dropped the second to last tx because it was too large to fit in the square.
		// The filtered builder should have dropped the last tx because the nonce for it was invalidated by dropping the second to last tx.
		require.Equal(t, 7, len(prepareResponse.Txs))
```